Lib/multiprocessing/resource_tracker.py

```diff
@@ -285,7 +292,13 @@ def main(fd):
         with open(fd, 'rb') as f:
             for line in f:
                 try:
-                    cmd, name, rtype = line.strip().decode('ascii').split(':')
+                    cmd, enc_name, rtype = line.rstrip(b'\n').decode('ascii').split(':', 2)
+                    if rtype == "shared_memory":
+                        name = base64.urlsafe_b64decode(enc_name.encode('ascii')).decode('utf-8', 'surrogateescape')
```

Comment:
Decoding is OK with strings.
```suggestion
                        name = base64.urlsafe_b64decode(enc_name).decode('utf-8', 'surrogateescape')
```

Reply:
Thanks for your comment. I like your suggestion to consider using JSON to encode the message. 

Also, in the original implementation there was a check that the message is not longer than 512 bytes, since writes shorter than `PIPE_BUF` (512 bytes on POSIX) are guaranteed to be atomic. But there was no check on the name length.  

Now I check that the name is at most 255 bytes long, which is the value of `NAME_MAX` on Linux (including the leading slash that POSIX requires in shared memory and semaphore names).  

I still encode the name using Base64, because `json.dumps(..., ensure_ascii=True)` would otherwise expand each non-ASCII byte into a 6-character escape like `\uDC80`. Using Base64 ensures that a 255-byte name becomes at most 340 bytes long, so the total JSON message always remains well below 512 bytes.  

As a result, the previous runtime check for the message length is now replaced by an `assert`.

What do you think?